File: backend/migration/mobility-monolith/src/main/java/com/example/PlateformeMobilite/DTO/UserResponseDTO.java
package com.example.PlateformeMobilite.DTO;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserResponseDTO {
    private String id;
    private String firstname;
    private String lastname;
    private String placeOfBirth;
    private LocalDate dateOfBirth;
    private String nationality;
    private String cin;
    private String email;
    private String username;
    private Boolean enabled;
    private Boolean passwordNeedToBeModified;
    private LocalDateTime lastLogin;
    private String createBy;
    private LocalDateTime createdDate;
    private String lastModifiedBy;
    private LocalDateTime lastModifiedDate;
    private List<String> roles;
}

File: frontend/angular-app/src/app/dto/UserResponseDTO.ts
import { Gender } from './enums/Gender';

export interface UserResponseDTO {
  id: string;
  firstname: string;
  lastname: string;
  placeOfBirth: string;
  dateOfBirth: string;
  nationality: string;
  gender: Gender;
  cin: string;
  email: string;
  username: string;
  enabled: boolean;
  passwordNeedToBeModified: boolean;
  lastLogin: string;
  createBy: string;
  createdDate: string;
  lastModifiedBy: string;
  lastModifiedDate: string;
  roles: string[];
}
